test/utils.js

```diff
@@ -0,0 +1,21 @@
+const { detectType, esriTypeMap } = require('../src/utils')
+
+describe('Detect Types', () => {
+  describe('detectType', () => {
+    it('should return correct string representation of a values data type', () => {
+      detectType(1).should.equal('Integer')
+      detectType(100.1).should.equal('Double')
+      detectType('test').should.equal('String')
+      detectType(new Date()).should.equal('Date')
+    })
+  })
+
+  describe('esriTypeMap', () => {
+    it('should return correct string representation of a values data type', () => {
+      esriTypeMap('Integer').should.equal('esriFieldTypeInteger')
+      esriTypeMap('Double').should.equal('esriFieldTypeDouble')
+      esriTypeMap('String').should.equal('esriFieldTypeString')
+      esriTypeMap('Date').should.equal('esriFieldTypeDate')
```

Comment:
there are some other types we should potentially add coverage for http://desktop.arcgis.com/en/arcobjects/latest/java/api/arcobjects/com/esri/arcgis/geodatabase/esriFieldType.html

Reply:
Added in b15b98c613aa506bc19cbaa929d6d119a8c4f2d3.

Reply:
Update as per review comments in c0cc0f1a7d85247305f1c808479525dfc486bed5